images/win/scripts/Installers/Install-Ruby.ps1

```diff
@@ -0,0 +1,144 @@
+################################################################################
+
+function Get-RubyVersions
+{
+    param (
+        [System.String] $Arch = "x64",
+        [System.String] $Extension = "7z"
+    )
+
+    $uri = "https://api.github.com/repos/oneclick/rubyinstaller2/releases"
+    try
+    {
+        $versionLists = @{}
+        $assets = (Invoke-RestMethod -Uri $uri).Where{ -not $_.prerelease }.assets
+        $7zArchives = $assets.Where{ $_.name.EndsWith("$Arch.$Extension") }
+        $majorMinorGroups = $7zArchives | Group-Object { $_.name.Replace("rubyinstaller-", "").Substring(0, 3) }
+        foreach($majorMinorGroup in $majorMinorGroups)
+        {
+            $group = $majorMinorGroup.Group
+            $sortVersions = $group | Sort-Object {[Version]$_.name.Replace("rubyinstaller-", "").Replace("-$Arch.$Extension","").Replace("-",".")}
+            $latestVersion = $sortVersions | Select-Object -Last 1
+            $versionLists[$majorMinorGroup.Name] = $latestVersion.browser_download_url
+
+        }
+        return $versionLists
+    }
+    catch
+    {
+        Write-Host "Enable to send request to the '$uri'. Error: '$_'"
+        exit 1
+    }
+}
+
+# Most of this logic is from
+# https://github.com/MSP-Greg/actions-ruby/blob/master/lib/main.js
+function Install-Ruby
+{
+    param(
+        [String]$PackagePath,
+        [String]$Architecture = "x64"
+    )
+
+    # Expand archive with binaries
+    $packageName = [IO.Path]::GetFileNameWithoutExtension((Split-Path -Path $PackagePath -Leaf))
+    $tempFolder = Join-Path -Path $env:TEMP -ChildPath $packageName
+    Extract-7Zip -Path $PackagePath -DestinationPath $env:TEMP
+
+    # Get Ruby version from binaries
+    $rubyVersion = & "$tempFolder\bin\ruby.exe" -e "print RUBY_VERSION"
+
+    if ($rubyVersion)
+    {
+        Write-Host "Installing Ruby $rubyVersion"
+        $rubyToolcachePath = Join-Path -Path $env:AGENT_TOOLSDIRECTORY -ChildPath "Ruby"
+        $rubyVersionPath = Join-Path -Path $rubyToolcachePath -ChildPath $rubyVersion
+        $rubyArchPath = Join-Path -Path $rubyVersionPath -ChildPath $Architecture
+
+        if (-not (Test-Path $rubyToolcachePath))
+        {
+            Write-Host "Creating Ruby toolcache folder"
+            New-Item -ItemType Directory -Path $rubyToolcachePath | Out-Null
+        }
+
+        Write-Host "Creating Ruby '${rubyVersion}' folder in '${rubyVersionPath}'"
+        New-Item -ItemType Directory -Path $rubyVersionPath -Force | Out-Null
+
+        Write-Host "Moving Ruby '${rubyVersion}' files to '${rubyArchPath}'"
+        Move-Item -Path $tempFolder -Destination $rubyArchPath | Out-Null
+
+        Write-Host "Removing Ruby '${rubyVersion}' documentation '${rubyArchPath}\share\doc' folder"
+        Remove-Item -Path "${rubyArchPath}\share\doc" -Force -Recurse -ErrorAction Ignore
+
+        Write-Host "Installing Bundler"
+        cmd.exe /c "$rubyArchPath\bin\gem.cmd install bundler -v ""~> 2"" --no-document"
```

Comment:
If ruby/setup-ruby reuses these installations in the future, `Bundler` should not be pre-installed, and the choice should be left to the user. The Bundler version used for `bundle ...` on the command line will be the latest version if there is no `Gemfile.lock` (otherwise, it will try to pick the same major version as in the Gemfile.lock).

Also Ruby 2.6 & 2.7 already come with their own Bundler as a default gem, and having Bundler 2.1.4 installed twice can lead to complications.

So, unless it's strictly required, I would leave installing Bundler to the user, which might want a specific version.

Reply:
Thank you. Removed.